### PR TITLE
Merge US request holdings into staging

### PR DESCRIPTION
**Pull request by vgranata**
- Add request of holding
Note: Temporary items are deleted once a day with a cronjob.

Co-Authored-by: Valeria Granata <valeria@chaw.com>

## Why are you opening this PR?

https://tree.taiga.io/project/rero21-reroils/us/1809?milestone=306074
Solves issue: https://github.com/rero/rero-ils/issues/2258

## Dependencies

My PR depends on the following `rero-ils-ui`'s PR(s):

* rero/rero-ils-ui/pull/777

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Cypress tests successful?

### CODE REVIEW

**Comment by pronguen** [comment]
- [ ] The request button is not displayed on holdings, neither in the public interface, nor in the professional interface. Therefore, it is impossible to test the US.

**Comment by ManaDeweerdt** [comment]
- [ ] Once checked out and returned, the items is not deleted.
- [ ]

**Comment by vgranata** [comment]
@zannkukai The request date seems to disappear from https://ilsdev.test.rero.ch/professional/circulation/requests with this commit:
https://github.com/rero/rero-ils-ui/commit/d6c739a6e5bcdbb6347b6331c0a5a2a5a6123c79

**Comment by zannkukai** [comment]
> @zannkukai The request date seems to disappear from https://ilsdev.test.rero.ch/professional/circulation/requests with this commit: [rero/rero-ils-ui@d6c739a](https://github.com/rero/rero-ils-ui/commit/d6c739a6e5bcdbb6347b6331c0a5a2a5a6123c79)

It should be solved with https://github.com/rero/rero-ils/pull/2646 integration. I merged to quickly the UI pr. Sorry XD

**Review by BadrAly — APPROVED** [review]

**Comment by PascalRepond** [comment]
- For holdings that contain issue-type items, the items do not appear in the public view ! The usual dropdown disappeared
image
- The request button does not appear for holdings that contain items (normal ?)
- The temporary item still seems to appear in the public view
image
- The temporary item is not deleted once it has been and returned

**Comment by vgranata** [comment]
REP: For holdings that contain issue-type items, the items do not appear in the public view ! The usual dropdown disappeared

If I'm not mistaken this problem appears with this commit https://github.com/rero/rero-ils-ui/commit/24ae5b405dcbec83295d882cd304013cfbbac0c3. 
@zannkukai if you replace 258 with 251 here: https://github.com/rero/rero-ils-ui/blob/24ae5b405dcbec83295d882cd304013cfbbac0c3/projects/public-holdings-items/src/index.html#L43 you will see that the dropdown menu disappears. Do you already have a PR to solve this problem ? thanks

**Comment by vgranata** [comment]
REP: The temporary item still seems to appear in the public view
This PR https://github.com/rero/rero-ils/pull/2692 should solve the problem.

**Comment by vgranata** [comment]
Once checked out and returned, the items is not deleted. 
Temporary items are deleted once a day with a cronjob.

**Review by pronguen — APPROVED** [review]

File: rero_ils/alembic/05555c03fe49_correct_holdings_items_count.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        item_search = ItemsSearch()[0:0]\
+                .filter('term', holding__pid=pid)\
+                .filter('term', issue__status="received")
+    else:
+        item_search = ItemsSearch()[0:0]\
+                .filter('term', holding__pid=pid)
+
+    # compute number of masked item
+    item_search.aggs.bucket('public_items', 'terms', field='_masked')
+    results = item_search.source(['organisation', 'library']).execute()
```
Comment (jma) [inline rero_ils/alembic/05555c03fe49_correct_holdings_items_count.py:92]: source seems not be used

File: rero_ils/alembic/05555c03fe49_correct_holdings_items_count.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    # number of items
+    items_count = results.hits.total.value
+    # number of masked items
+    number_of_masked_items = 0
+    for bucket in results.aggregations.public_items.buckets:
+        if bucket.key_as_string == 'true':
+            number_of_masked_items = bucket.doc_count
```
Comment (jma) [inline rero_ils/alembic/05555c03fe49_correct_holdings_items_count.py:99]: usually the bucket is not safe to get the count: iterate over all results (scan()) by getting only the masked property is safer.

**Review by jma — APPROVED** [review]

File: rero_ils/alembic/05555c03fe49_correct_holdings_items_count.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (vgranata) [reply rero_ils/alembic/05555c03fe49_correct_holdings_items_count.py:92]: The code in this file comes from:
https://github.com/rero/rero-ils/blame/staging/rero_ils/modules/holdings/listener.py
It was yourself to write it, are you sure you want me to change it ?
I guess I should change it in the listener too.

**Pull request merged**